sway_libs/src/nft/extensions/meta_data/meta_data.sw

```diff
@@ -0,0 +1,68 @@
+library meta_data;
+
+dep meta_data_structures;
+
+use meta_data_structures::NFTMetaData;
+use ::nft::{errors::InputError, nft_core::NFTCore, nft_storage::{META_DATA, TOKENS}};
+use std::{hash::sha256, storage::{get, store}};
+
+pub trait MetaData<T> {
+    /// Returns the metadata for this token
+    #[storage(read)]
+    fn meta_data(self) -> Option<T>;
+    /// Creates new metadata for this token.
+    ///
+    /// # Arguments
+    ///
+    /// * `metadata` - The metadata value to be included in the new metadata.
```

Comment:
After looking at the implementation "included" isn't quite the right word imo.  IMO this line should read "The new metadata value to overwrite the old value." or something like that